### PR TITLE
Introduce new convention for project/app naming

### PR DESCRIPTION
**Pull request by davydovanton**
Reference #499

What will happen after calling `hanami new MyApp`:
screenshot 2016-02-08 20 05 40

### CODE REVIEW

File: lib/hanami/application_name.rb [hunk 1]
```diff
@@ -28,7 +30,7 @@ class ApplicationName
     #
     # @since 0.2.1
     def initialize(name)
-      @name = sanitize(name)
+      @name = sanitize(Hanami::Utils::String.new(name).underscore)
```
Comment (davydovanton) [inline lib/hanami/application_name.rb]: what do you think, maybe 'Utils' module is better place for `#sanitize` method?

**Comment by joneslee85** [comment]
@davydovanton let's hold off the implementation until we reach a consensus

**Comment by davydovanton** [comment]
@joneslee85 okay. I agree with you :+1:

**Comment by jodosha** [comment]
@joneslee85 @davydovanton The more I see these PRs fixing bugs with naming conventions, the more I think we should extract this logic into an internal object like `Hanami::ComponentName` (sorry for the poor name :sweat_smile:) and use it for all the code generators.

**Comment by davydovanton** [comment]
@jodosha what do you think may be is better to close this PR?

**Comment by jodosha** [comment]
@davydovanton Yes, I think this is fixed now. Thanks! :+1:

**Pull request closed without merging**